plasmapy/utils/data/downloader.py

```diff
@@ -4,74 +4,409 @@
+
+    @property
+    def _api_connected(self) -> bool:
+        """
+        Returns `True` if a connection exists to the API, otherwise `False`.
+        """
+        try:
+            # Requesting this URL does not count as an API query
+            self._http_request("https://api.github.com/rate_limit")
+        # No testing because CI always has a connection to the API
+        except requests.ConnectionError:  # coverage: ignore
+            return False
+        return True
+
+    @property
+    def _api_usage(self) -> tuple[int, int]:
+        """
+        Return the API call limit and the number currently used from this IP.
+        """
+        # Ensure that the GitHub API is not rate limited
+        reply = self._http_request("https://api.github.com/rate_limit")
+        info = reply.json()
+        rate_info = info["resources"]["core"]
+        limit = int(rate_info["limit"])
+        used = int(rate_info["used"])
+
+        return limit, used
+
+    @property
+    def _api_is_rate_limited(self) -> bool:
+        """
+        Whether or not the API is currently rate limited.
+        """
+        limit, used = self._api_usage
+        return used >= limit
+
+    @property
+    def _do_validation(self) -> bool:
+        """
+        Determine whether or not to enforce validation using the GitHub API.
+        """
+        return self._validate and self._api_connected and not self._api_is_rate_limited
+
+    def _update_repo_blob_dict(self) -> None:
+        """
+        Update the blob file with a call to the repository.
+
+        Raises
+        ------
+        ValueError
+            If the URL does not return the expected JSON file with the
+            expected keys.
+
+        Returns
+        -------
+        repo_blob_dict : dict
+            Dictionary with filenames as keys. Each item is another entry
+            with keys `sha` and `download_url`.
+        """
+        # If the current blob file has been updated in the past 5 minutes,
+        # don't bother doing it again
+        # Ignore in tests, as this won't happen in CI
+        try:
+            if time.time() - self._blob_dict["_timestamp"] < 300:
+                return None  # coverage : ignore
+        # If the _timestamp key hasn't been set yet, the blob file has
+        # never been updated before
+        except KeyError:  # coverage : ignore
+            pass
+
+        # If this instance of Downloader has already updated from the API once,
+        # don't do it again. Almost certainly nothing has changed!
+        # Not tested, since CI never waits >5 min with the same Downloader
+        # instantiated.
+        if self._updated_blob_file_from_repo:  # coverage: ignore
+            return None
+
+        reply = self._http_request(self._API_BASE_URL)
+
+        # Extract the SHA hash and the download URL from the response
+
+        # Extract contents to JSON
+        # Not tested, since any URL on the gituhb API that doesn't raise a 404
```

Comment:
```suggestion
        # Not tested, since any URL on the GitHub API that doesn't raise a 404 error
```
Incidentally, 404 was the number of a differential equations class I took!  Either that, or I never managed to find the course website. 😬